src/lib/utils.ts

```diff
@@ -143,10 +143,13 @@ export function validateMetadata(object: Package, name: string): Package {
 export function combineBaseUrl(protocol: string, host: string | void, prefix?: string | void): string {
   let result = `${protocol}://${host}`;
 
-  if (prefix) {
-    prefix = prefix.replace(/\/$/, '');
+  if (prefix && prefix.length !== 0) {
+    prefix = prefix.trim();
+    if (prefix.endsWith('/')) {
+      prefix = prefix.slice(0, -1);
+    }
 
-    result = prefix.indexOf('/') === 0 ? `${result}${prefix}` : prefix;
+    result = prefix.length === 0 || prefix.indexOf('/') === 0 ? `${result}${prefix}` : prefix;
```

Comment:
```suggestion
    result = prefix.length === 0 || prefix.startsWith('/') ? `${result}${prefix}` : prefix;
```

Reply:
All too complicated, also ternary operators are not great to read. See my proposed alternative solution.

https://github.com/verdaccio/verdaccio/pull/1647#issuecomment-571601242